ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1010,38 +1041,37 @@ public void write() throws StoreException {
       try {
         if (indexes.size() > 0) {
           // before iterating the map, get the current file end pointer
-          Map.Entry<Long, IndexSegment> lastEntry = indexes.lastEntry();
+          Map.Entry<Offset, IndexSegment> lastEntry = indexes.lastEntry();
```

Comment:
Yes, there is going to be an `AtomicReference` to make sure the persistence works correctly.